### PR TITLE
Move Log4j2 backend out of `flogger` modules

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR moves `flogger-log4j2-backend` to `backends` directory, renaming it to `logging-log4j2-backend`.

The name of the published artifact will be `spine-logging-log4j2-backend`.  I avoided usage of `jvm` prefix (`logging-jvm-log4j2-backend`) because it results in too long artifact name. And it seems redundant because it is very unlikely that we would have two same-named backends for different targets.

Package location and copyright notices have also been updated.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL

File: backends/logging-log4j2-backend/src/main/java/io/spine/logging/backend/log4j2/package-info.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Contains implementation of `Log4j2` backend.
```
Comment (armiol) [inline backends/logging-log4j2-backend/src/main/java/io/spine/logging/backend/log4j2/package-info.java:28]: Let's `{@code ...}`.

**Review by armiol — APPROVED** [review]
@yevhenii-nadtochii LGTM with a minor comment.

File: backends/logging-log4j2-backend/src/main/java/io/spine/logging/backend/log4j2/Log4j2BackendFactory.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    // Implementation exposes more methods that are needed by the backend.
+    // So, we have to cast an interface back to its implementation.
+    var logger = (Logger) LogManager.getLogger(name);
+
+    return new Log4j2LoggerBackend(logger);
+  }
+
+  @Override
+  public String toString() {
+    return "Log4j2 backend";
```
Comment (alexander-yevsyukov) [inline backends/logging-log4j2-backend/src/main/java/io/spine/logging/backend/log4j2/Log4j2BackendFactory.java:67]: Let's use fully-qualified string name instead. If not in this PR, we'd better have it someday. There was also a comment from original authors on a similar matter in one class from the previous PR.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with one suggestion for future PRs.

**Pull request merged**